specs/bond-manager.md

```diff
@@ -0,0 +1,100 @@
+bonds will not be required since the attestors are permissioned. But in more complex games, such as the
+fault dispute game, bonds will be required since the challengers and defenders perform a series of
+alternating onchain transactions requiring bonds at each step. In this case, a separate bond manager than
+that of the `L2OutputOracle` is required.
+
+## The Bond Problem
+
+At its core, the bond manager is straightforward - it escrows or holds ether and returns
+it when finalized or seized. But the uncertainty of introducing bonds lies in the
+bond _sizing_, i.e. how much should a bond be? Sizing bonds correctly entails a variety of
+tradeoffs. Price them too high, and not enough proposers will post outputs. Price them too low, and
+challengers won't be incentivized to delete outputs if the gas cost of doing so outways the bond itself.
+
+Below, we outline two different approaches to sizing bonds and the tradeoffs of each.
+
+### Simple Bond
+
+The _Simple Bond_ is a very conservative approach to bond management, establishing a **fixed** bond
+size using the worst case gas cost for deleting an output proposal. The idea being
+that a bond posted for a given output proposal must at least cover the
+cost of the challenge agents deleting that output proposal in order to make
+dispute games incentive compatible.
+
+With this approach, the Simple Bond is fixed to `1 ether`. By working backwards, we
+can establish that the cost of challenging an output proposal must not
+exceed `1 ether`. As such, this implies a base fee of `10,000` for a challenge costing
+`100,000` gas with a `1 gwei priority fee`. This leaves challenger agents with a significant
+buffer between the historical highest gas price of roughly `236` in 2020, and the base gas fee of `10,000`.
+
+### Variable Bond
+
+Better bond heuristics can be used to establish a bond price that accounts for
+the time-weighted gas price. One instance of this called _Varable Bonds_ use a separate oracle contract,
+`GasPriceFluctuationTracker`, that tracks gas fluctuations within a pre-determined
+bounds. This replaces the ideal solution of tracking challenge costs over all L1
+blocks, but provides a reasonable bounds. Proposers are responsible for funding this contract.
+
+## Types of Bond Managers
+
+Below we outline the bond manager contracts.
+
+### OracleBondManager
+
+The oracle bond manager is handles bond management on behalf of the `L2OutputOracle`.
+
+In the `L2OutputOracle`, a permissionless proposer can post an output proposal, which
+becomes finalized after a pre-determined interval detailed in the
+[proposals doc](./proposals.md). When proposing an output, the proposer must
+post a bond that is used to disincentivize spam. When the output is finalized,
+the bond may be claimed by the proposer. But in the case that the output is invalid,
+the bond *should* be seized by a set of challenge agents or a
+[dispute game](./dispute-game.md).
+
+### AttestationBondManager
```

Comment:
I think we can focus on having a single type of bond manager for now and say that a new type of bond manager will be introduced in the future to enable the challenge response nature of the fault proof dispute game

Reply:
Agreed 👍